core/src/main/scala/kafka/log/ProducerIdMapping.scala

```diff
@@ -261,7 +261,7 @@ class ProducerIdMapping(val config: LogConfig,
         case None =>
           lastSnapOffset = 0L
           lastMapOffset = 0L
-          snapDir.mkdir()
+          Files.createDirectories(snapDir.toPath)
```

Comment:
The reason we do it here is that a running broker may get a partition assigned to it at any moment, at which point the pid-mapping sub directory will need to be created.